### ISSUE TITLE
Bug: too many shader compilations

### ISSUE THREAD

**Issue by munrocket** [compute-toys/compute.toys#45]
check dev console on page load.

**Comment by davidar** [comment]
Yeah, known issue but it's a tricky one to fix, it's due to React triggering updates as it loads all the components. Maybe we could suppress shader compiles until the page has fully loaded? But I'm not entirely sure how to implement that. In any case webgpu should be caching the compiled shaders, so doesn't seem to be a huge issue in practice, though still not ideal

**Comment by munrocket** [comment]
Here they fixed reconciliation https://github.com/pmndrs/its-fine
Maybe we can just use it here?
